src/MySqlConnector/UnixEndPoint.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+
+namespace MySql.Data
+{
+	// copied from https://github.com/mono/mono/blob/master/mcs/class/Mono.Posix/Mono.Unix/UnixEndPoint.cs
```

Comment:
It's probably appropriate to copy the full copyright and license statement from the original file, too.